packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -204,6 +204,9 @@ class DataSourceWithBackend<
     const headers: Record<string, string> = {};
     headers[PluginRequestHeaders.PluginID] = Array.from(pluginIDs).join(', ');
     headers[PluginRequestHeaders.DatasourceUID] = Array.from(dsUIDs).join(', ');
+    if (request?.originalQueryDelta !== undefined) {
+      headers['X-Grafana-Modified-From'] = request.originalQueryDelta.toString(10);
```

Comment:
The hope is that we can use this in Faro or in Grafana logs to generate a dashboard displaying how many records we've avoided querying

Reply:
thinking more about this, let's try a different approach that will hopefully be more easily consumed by leveraging Faro (we can probably do this in a follow-up to keep this PR manageable)

basically, we should try to measure the amount of data and latency during the initial full-range request. we can do this by timing `fetch()` and inspecting the `Content-Length` and `Content-Encoding` headers of the response.

we can then assume that every full-range refresh would have been the same size and the same latency, it may not always be true, but i think it is in most cases. but when we do an incremental request instead, we can take the difference and calculate the "savings" in latency and in transfer size, and send this savings info to Faro. it won;t be exact, but i think it will be close enough.

Reply:
pushed https://github.com/grafana/grafana/commit/0090d6f11b673c0d45e0cf26343c211f225f749b but it looks like the backend does not currently send `Content-Length` headers, so we don't get anything here:

https://github.com/grafana/grafana/commit/0090d6f11b673c0d45e0cf26343c211f225f749b#diff-a38b9903e300e119fb4b391c6bd7332390d562c5c2215e1e6a68afe504e8117bR11